### ISSUE TITLE
rightPop(100, MILLISECONDS) translates to "BRPOP key, 0" [DATAREDIS-176]

### ISSUE THREAD

**Issue by spring-projects-issues** [spring-projects/spring-data-redis#754]
**[Dmitry Sidorenko](https://jira.spring.io/secure/ViewProfile.jspa?name=ushkinaz)** opened **[DATAREDIS-176](https://jira.spring.io/browse/DATAREDIS-176?redirect=false)** and commented

When given timeout in list operations is less than a second, Spring-data rounds it to 0. Zero means "block indefinitely" in redis.

Probably, spring should never round to zero. Correct behaviour would be something like this:

0 ms => 0s
100 ms => 1 s
1000 ms => 1 s
1010 ms => 1 s
1900 ms => 2 s

In other words, for anything lower than 1 sec, but not 0, round to 1 sec. For everything else - use TimeUnit#toSeconds

---

**Affects:** 1.0.4

**Referenced from:** commits https://github.com/spring-projects/spring-data-redis/commit/7c8dafce672d6b5a7fe8e565e825cfa77878ba37

**Backported to:** [1.0.5](https://github.com/spring-projects/spring-data-redis/milestone/11?closed=1)
